tests/net/unix.rs

```diff
@@ -208,21 +190,10 @@ fn test_unix_msg() {
                 )
                 .unwrap();
             }
-
-            unlinkat(cwd(), path, AtFlags::empty()).unwrap();
         }
     };
 
     let client = move || {
-        {
-            let (lock, cvar) = &*ready;
-            let mut started = lock.lock().unwrap();
-            while !*started {
-                started = cvar.wait(started).unwrap();
-            }
-        }
```

Comment:
I'm curious why the condition variable code is removed here. This is meant to ensure that the client waits for the server to be ready; is this no longer needed?

Reply:
See the commit message for that commit: https://github.com/bytecodealliance/rustix/pull/661/commits/b5e97708f9f6f0ae9a57965ea3cb4779827fcd22

> test_unix_msg: Remove thread synchronisation
>
> This refactors the test_unix_msg() test a bit. No change in behaviour is
> intended.
>
> Previously, two threads were started in parallel. The server thread used
> a mutex and a condition variable to indicate that it set up its
> listening socket. The client thread waited for this signal before it
> attempted to connect.
> 
> This commit makes the code instead bind the server socket before
> starting the worker threads. That way, no synchronisation is necessary.

The actual reason why I did this was that at first I passed in the already-connected socket to the new helper function. That would have been a lot more complicated with this synchronisation mechanism. Only later did I notice that I only need to pass in the socket address and let the helper set up the socket itself.

Still, binding the socket before the "worker threads" are started seems like a simplification to me that gets rid of some unnecessary code. AFAIR there should not be any difference in which thread creates the socket and whether the socket is moved to another thread.